tenseal/cpp/tensors/ckkstensor.cpp

```diff
@@ -515,9 +515,14 @@ shared_ptr<CKKSTensor> CKKSTensor::_dot_inplace(
             this->sum_inplace();
             return shared_from_this();
         } else if (other_shape.size() == 2) {  // 1D-2D
-            // TODO: better implement broadcasting for mul first then would be
-            // implemented similar to 1D-1D
-            throw invalid_argument("1D-2D dot isn't implemented yet");
+            if (this_shape[0] != other_shape[0])
```

Comment:
Why isn't valid for 
```
this_shape[0] == other_shape[1]
```
?

Reply:
We need the vector size to match the first dimension of the matrix to perform the dot product